pxr/usd/sdf/childrenView.h

```diff
@@ -188,23 +187,102 @@ class SdfChildrenView {
     // the owner's storage. That allows the iterator to operate without
     // knowing anything about the specific data storage that's used,
     // which is important for providing both Gd and Lsd backed storage.
-    class _InnerIterator :
-        public boost::iterator_facade<_InnerIterator,
-                                      value_type,
-                                      std::random_access_iterator_tag,
-                                      value_type> {
+    class _InnerIterator {
     public:
-        typedef value_type reference;
-        typedef size_t size_type;
-        typedef ptrdiff_t difference_type;
+        using iterator_category = std::random_access_iterator_tag;
+        using value_type = SdfChildrenView::value_type;
+        using reference = value_type;
+        using pointer = void;
+        using difference_type = std::ptrdiff_t;
 
-        _InnerIterator() :
-            _owner(NULL), _pos(0) { }
+        _InnerIterator() = default;
         _InnerIterator(const This* owner, const size_t& pos) :
             _owner(owner), _pos(pos) { }
 
```

Comment:
I'm pretty sure you're going to need operator-> here too. This iterator is the type for SdfChildrenView::const_iterator (after all the template expansion) so it's highly likely that we have Presto code that dereferences via ->